code/debugging/src/debug_logger.py

```
@@ -0,0 +1,160 @@
+        listener (Listener): Listener to run
+    """
+    running = True
+    while running:
+        try:
+            conn = listener.accept()
+            print(f"[debug_logger]: connection accepted from {listener.last_accepted}")
+            msg = None
+            if conn.poll(timeout=2.0):
+                msg = conn.recv()
+                if isinstance(msg, str):
+                    if msg.lower() == CLOSE_MSG:
```

Ruff

Use a single `if` statement instead of nested `if` statements

Combine `if` statements using `and`

(SIM102)